### PR TITLE
Do not use the precompile address 0x05 in the wallet tests

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/contracts/Wallet.cpp [hunk 1]
```diff
@@ -559,24 +559,24 @@ BOOST_AUTO_TEST_CASE(multisig_value_transfer)
 	BOOST_REQUIRE(callContractFunction("addOwner(address)", h256(account(3), h256::AlignRight)) == encodeArgs());
 	// 4 owners, set required to 3
 	BOOST_REQUIRE(callContractFunction("changeRequirement(uint256)", u256(3)) == encodeArgs());
-	// check that balance is and stays zero at destination address
-	BOOST_CHECK_EQUAL(balanceAt(Address(0x05)), 0);
+	Address destination = account(5);
+	auto destinationBalance = balanceAt(destination);
```
Comment (axic) [inline test/contracts/Wallet.cpp]: Which one is better? This option or sending off all the balance first (see below).

Comment (axic) [reply test/contracts/Wallet.cpp]: Oh never mind, only the first account ever created gets a balance in RPCSession.

Comment (axic) [reply test/contracts/Wallet.cpp]: Probably better creating a helper `emptyAccount` which forces creation of a new one.

File: test/RPCSession.cpp [hunk 2]
```diff
@@ -339,13 +339,16 @@ Json::Value RPCSession::rpcCall(string const& _methodName, vector<string> const&
...
+}
+
 string const& RPCSession::accountCreateIfNotExists(size_t _id)
 {
-	if (_id >= m_accounts.size())
-	{
-		m_accounts.push_back(personal_newAccount(""));
-		personal_unlockAccount(m_accounts.back(), "", 100000);
-	}
+	while (_id > m_accounts.size())
```
Comment (axic) [inline test/RPCSession.cpp]: Basically this was buggy before, if say there were accounts 1, 2, 3 created and 5 was requested, then it was out of bounds access.

File: test/RPCSession.cpp [hunk 3]
```diff
@@ -339,22 +339,25 @@ Json::Value RPCSession::rpcCall(string const& _methodName, vector<string> const&
 	return result["result"];
 }
 
+string const& RPCSession::accountCreate()
```
Comment (chriseth) [inline test/RPCSession.cpp]: Trust the force you must?

File: test/RPCSession.h [hunk 4]
```diff
@@ -121,6 +121,7 @@ class RPCSession: public boost::noncopyable
 	Json::Value rpcCall(std::string const& _methodName, std::vector<std::string> const& _args = std::vector<std::string>(), bool _canFail = false);
 
 	std::string const& account(size_t _id) const { return m_accounts.at(_id); }
+	std::string const& accountCreate();
```
Comment (chriseth) [inline test/RPCSession.h]: Ah I see, consistent it is.

File: test/contracts/Wallet.cpp [hunk 5]
```diff
@@ -458,6 +458,11 @@ class WalletTestFramework: public SolidityExecutionFramework
 		sendMessage(*s_compiledWallet + args, true, _value);
 		BOOST_REQUIRE(!m_output.empty());
 	}
+
+	Address emptyAccount()
+	{
+		return Address(m_rpc.accountCreate());
```
Comment (chriseth) [inline test/contracts/Wallet.cpp]: You know that this takes about  a second to run?

**Review by chriseth — APPROVED** [review]

File: test/RPCSession.cpp (hunk 3 above)
Comment (axic) [reply test/RPCSession.cpp]: Followed the naming of the other one, `accountCreateIfNotExists`.

File: test/contracts/Wallet.cpp (hunk 5 above)
Comment (axic) [reply test/contracts/Wallet.cpp]: There's the alternative solution in commit one. Handling the starting balance.

Comment (axic) [reply test/contracts/Wallet.cpp]: @chriseth so which one should we have:
1) take starting balance and compare to that
2) send off all the balance to somewhere (ie 0)
3) create empty account

The problem with 2) is that `sendEther` doesn't currently enforces it was successful, though we can add another check to see if the balance is 0 after.

Comment (chriseth) [reply test/contracts/Wallet.cpp]: 4. just use a different address than 5?

File: test/contracts/Wallet.cpp [hunk 6]
```diff
@@ -559,24 +559,24 @@ BOOST_AUTO_TEST_CASE(multisig_value_transfer)
...
-	// check that balance is and stays zero at destination address
-	BOOST_CHECK_EQUAL(balanceAt(Address(0x05)), 0);
+	Address destination = Address("0x5c6d6026d3fb35cd7175fd0054ae8df50d8f8b41");
+	BOOST_CHECK_EQUAL(balanceAt(destination), 0);
 	m_sender = account(0);
 	sendEther(account(1), 10 * ether);
 	m_sender = account(1);
-	auto ophash = callContractFunction("execute(address,uint256,bytes)", h256(0x05), 100, 0x60, 0x00);
-	BOOST_CHECK_EQUAL(balanceAt(Address(0x05)), 0);
+	auto ophash = callContractFunction("execute(address,uint256,bytes)", h256(destination), 100, 0x60, 0x00);
```
Comment (axic) [inline test/contracts/Wallet.cpp]: Hm, why were these `h256`?

**Comment by axic** [comment]
Strangely with fixed address it doesn't work:

```
/home/travis/build/ethereum/solidity/test/contracts/Wallet.cpp(579): error in "multisig_value_transfer": check balanceAt(destination) == 100 failed [0 != 100]
/home/travis/build/ethereum/solidity/test/contracts/Wallet.cpp(649): error in "revoke_transaction": check balanceAt(destination) == 100 failed [0 != 100]
/home/travis/build/ethereum/solidity/test/contracts/Wallet.cpp(690): error in "daylimit": check balanceAt(destination) == 90 failed [0 != 90]
```

**Pull request merged**